nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/main/java/org/nuxeo/ecm/platform/picture/core/im/IMImageUtils.java

```diff
@@ -71,14 +71,15 @@ public Blob call(Blob blob, String targetExt, String commandName) {
                 Framework.trackFile(targetFile, targetBlob);
                 return targetBlob;
             } catch (CommandNotAvailable | CommandException | IOException e) {
-                log.error("ImageMagick failed on command: " + commandName, e);
+                log.error(() -> String.format("ImageMagick failed on command: %s", commandName), e);
```

Comment:
Simpler: `log.error("ImageMagick failed on command: {}", () -> commandName), e)`
Same below.
But actually for an error log, there's not much point in trying to gain a millisecond, the original was fine.

Reply:
- `'log.error("ImageMagick failed on command: {}", () -> commandName, e)`: Perhaps i am missed something, but i tried (before the fix) for this syntaxe   but unfortunately it doesn't compile.

- _But actually for an error log, there's not much point in trying to gain a millisecond, the original was fine._ : The change wasn't related to time optimisation, the idea is to take advantage of the cleanup/format commit to change the log4j to log4j2 and use the varargs, it's habitually with Kevin for example  :)

Reply:
Ah yes if you have a supplier you can't pass an exception last. Then just `log.error("ImageMagick failed on command: {}", commandName, e)` which is actually better than my original message, and what I meant.

Reply:
_you have a supplier you can't pass an exception last_ exactly. yep for your proposal.

Reply:
In fact, you can : `log.error("ImageMagick failed on command: {}", () -> commandName, () -> e)`.
But in this case indeed there is no point.